integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -121,6 +121,7 @@ async function addSubscriptionAndSync<T>(
 }
 
 describe.skipIf(environment.missingServer, "Flexible sync", function () {
+  this.timeout(25 * 1000);
```

Comment:
To be more specific, setting this would override [this](https://github.com/realm/realm-js/blob/master/.github/workflows/pr-realm-js.yml#L191) 60000 ms timeout.

Reply:
Yeah having the longer timeout be overridden is not ideal. If we could fix that it would be great :) In the mean time (for this PR), do you think we should increase the timeout for this specific test suite to be the same as the `MOCHA_REMOTE_TIMEOUT`?

Reply:
For now that's probably the pragmatic answer.  Maybe @kraenhansen has an opinion.

Reply:
I can do a PR with an attempt to provide a reusable solution for this.
Basically I'd suggest adding `shortTimeout` and `longTimeout` as variables on the Mocha Remote context (available through the `environment` global) and provide some sane defaults to these (in the case of running locally without CI setting them). Then tests would be able to simply:

```
this.timeout(environment.longTimeout);
```

Reply:
Interesting idea.  We could also wrap that in a function to make it less breakable (`this.longTimeout()`)

Reply:
I like the combination of that, having `this.longTimeout()` be responisble for calling `this.timeout()` with the correct env 👍